lucene/core/src/java/org/apache/lucene/store/ByteBufferIndexInput.java

```diff
@@ -491,4 +492,14 @@ protected ByteBufferIndexInput buildSlice(String sliceDescription, long ofs, lon
       return super.buildSlice(sliceDescription, this.offset + ofs, length);
     }
   }
+
+  @Override
+  public boolean load() {
+    for (ByteBuffer buffer : buffers) {
+      if (buffer instanceof MappedByteBuffer) {
+        ((MappedByteBuffer) buffer).load();
```

Comment:
This would load more than necessary in the case of a slice. If you look at ByteBufferIndexInput#buildSlice, the offset of the first ByteBuffer might not be updated, so calling load() this way might load up to 1GB of data in main memory that we don't need. This might be especially problematic in the case of CFS files where all index inputs are slices of a single large file. We should maybe override this is MultiBufferImpl to ignore the first `offset` bytes of the first slice?

Reply:
good point I will look into this. I might need to think more about testing. I am not happy with the status quo.

Reply:
I pushed changes to address this. I still need to look into testing more.